### DISCUSSION TITLE
pybricksdev and pybricks on raspberry pi (connection ok, but nothing more)

### DISCUSSION THREAD

**Discussion by jajawunderbar** [pybricks/support#1923] [Q&A]
Hi, i use a raspberry pi 5 with new version of pybricksdev an pybricks (installed with venv and pip). The connection is running, but i cant start a motor or read the tiltsensor. no errormessage. using thonny. Here a bit of code. Thanks! Mathias

```
import asyncio
from pybricksdev.connections.pybricks import PybricksHub
from pybricksdev.ble import find_device
from pybricks.pupdevices import Motor as HubMotor
from pybricks.parameters import Port
from time import sleep

async def control_hub_motor():
    hub = PybricksHub()
    device = await find_device('Pybricks Hub')
    await hub.connect(device)
    print("Technic Hub erfolgreich verbunden")
    hub_motor_a = HubMotor(Port.A)
    hub_motor_a.dc(50)
    sleep(1)
    hub_motor_a.stop()
    await hub.disconnect()
    print("Verbindung zum Hub getrennt.")

asyncio.run(control_hub_motor())
```

**Comment by dlech** [comment]
> from pybricks.pupdevices import Motor as HubMotor
> from pybricks.parameters import Port

> print("Technic Hub erfolgreich verbunden")
>     hub_motor_a = HubMotor(Port.A)
>     hub_motor_a.dc(50)
>     sleep(1)
>     hub_motor_a.stop()

This needs to be placed in a separate `.py` file. Pybricks scripts run on the hub itself, not on the RPi.

Then use `hub.run("/path/to/file.py")` to run download and run the script on the hub.

> sleep(1)

We don't have this in Pybricks. Use `from pybricks.tools import wait` instead.

**Reply by jajawunderbar** [reply]
thanks! perfect!
